docs/architecture/adr-006-hermes-v0.2-usecases.md

```diff
@@ -0,0 +1,249 @@
+where users could get stuck. For example, the output of a command may be
+unclear, or there may be an error and thereby some CLI command
+finishes partially, or two relayers concurrently try to perform some
+operation(s) and interfere with each other, resulting in a chain state that is
+obscure to the user, and then the user could consequently be stuck.
+
+The first of the patterns below seeks to help "unblock" a user.
+The second pattern is a variation on the first; this permits more efficiency
+because it allows the reuse of previously-created objects in the
+creation of new objects on a chain (e.g., reuse a client in the creation of a
+connection, or reuse a connection in the creation of a new channel).
+
+#### Patterns
+
+We propose two basic patterns that Hermes should be able to fulfil.
+
+1. Simple invocations to perform basic actions.
+    - By _action_ here we mean doing the complete handshake for an object from
+      scratch (specifically _connection_ or _channel_) on two chains, or
+      relaying packets between two chains.
+    - The focus here is for the command to include retrying mechanisms 
+      (perform it _robustly_) and have the simplest interface.
+
+2. Allow reusing of pre-existing state for basic commands.
+    - The pre-existing state could be a client with some specific trust options,
+      for instance, and in this case Hermes would provide support for creating
+      a connection that uses this specific client.
+    - This pattern should also include a retrying mechanism.
+
+#### Details of Use-Cases
+
+Applying the above patterns to a few cases, we get the following concrete
+commands that Hermes v0.2.0 should fulfil.
+
+##### Create & Update a Client
+
+- Minimal invocation: this will create the client from scratch:
+
+```
+create client ibc-0 ibc-1 [--clock-drift <millis>] [--trusting-period <days>] [--trust-threshold <numerator/denominator>]
+```
+
+**Details:**
+Submits a transaction of type "client create" to chain `ibc-0` (called the
+_destination_ chain). The new client will be verifying headers for chain
+`ibc-1` (called a _source_ chain).
+
+- Update a client:
+
+```
+update client ibc-0 <client-id>
+```
+**Details:**
+Submits a transaction to chain `ibc-0` to update the client having identifier
+`<client-id>` with new consensus state. Hermes will automatically infer
+the source chain of this client from the [client state][client-state].
+
+##### Create New Connection
+
+- Minimal invocation: this will create the connection from scratch, using
+  _new_ clients:
+
+```
+create connection ibc-0 ibc-1 [--delay <delay>]
+```
+
+**Details:**
+Starts a transaction to perform the connection open handshake protocol between
+chains `ibc-0` and `ibc-1`. The optional parameter `--delay` is the delay period
+that the new connection should have.
+
+- Reusing pre-existing state, concretely, with _existing_ clients:
+
+```
+create connection ibc-0 --src-client <client-id> --dst-client <client-id> [--delay <delay>]
```

Comment:
Wondering if we should allow only one of the clients to be specified and the other created.

Reply:
This seems debatable. But this use-case should not require much change from the point of view of the CLI: except that the `--dst-client <client-id>` option can be missing (and if it's missing, then we create the client on the destination chain).

Reply:
Right. I was just wondering if it makes sense at all to allow this. We can leave this out for now and wait for more feedback.